src/pudl/transform/classes.py

```diff
@@ -431,6 +431,26 @@ def correct_units(df: pd.DataFrame, params: UnitCorrections) -> pd.DataFrame:
 #####################################################################################
 # Abstract Table Transformer classes
 #####################################################################################
+def cache_df(key: str = "main") -> Callable:
+    """A decorator for managing intermediate dataframe caching."""
+
+    def _decorator(func) -> Callable:
+        @wraps(func)
+        def _wrapper(self: AbstractTableTransformer, *args, **kwargs) -> pd.DataFrame:
+            df = func(self, *args, **kwargs)
+            if self.cache_dfs:
+                logger.debug(
+                    f"{self.table_id.value}: Caching df to {key=} "
+                    f"in {func.__name__}()"
+                )
+                self._cached_dfs[key] = df.copy()
```

Comment:
why a copied version of the df?

Reply:
So that subsequent alterations of the dataframe as it continues to be transformed can't alter the cached copy.